### PR TITLE
[iOS] Fix for Editor scaling creating new lines on older iOS versions

### LINKED ISSUE #25581: Editor Scaling creating new lines on older iOS versions
### Description

The issue is when an editor is scaled down on older iOS versions, the text splits in the middle and moves on to the next line causing text to seem to disappear.

This issue was believed to be fixed with: https://github.com/dotnet/maui/pull/24859.
Removing the call to SizeThatFits seemed to not cause this scaling issue but it turns out this is only on iOS 17 and up.

On iOS 15.5, the bad editor scaling behavior is still present and on iOS 16.4, it seems that there is still slightly different, but wrong behavior.

### Some things I've tried
* I removed all references to the Placeholder label to see if that subview was doing anything strange and it does not look like this has any effect. Also removing the other calls in LayoutSubviews to UpdatePlaceholderLabelFrame() and ShouldCenterVertically() do not affect this behavior which leads me to believe this issue is happening on the backend of things.
* It seems like the issue may be related to the ContentSize and the Bounds when scaling happens and these could perhaps be differences in how the different iOS versions handle it?

Inside the [MauiTextView.LayoutSubviews](https://github.com/dotnet/maui/blob/main/src/Core/src/Platform/iOS/MauiTextView.cs#L116-L122) method, I am setting a breakpoint and scaling down the editor by 0.5 and then scaling it back up to 1.0. It appears that there are two passes in the LayoutSubviews call and I capture the Bounds and ContentSize. The first image below shows iOS 17.0 and the second shows iOS 15.5.  You'll even see when the editor on iOS 15.5 is at the full scale and appears on one line, it still has a much larger ContentSize.Height than the iOS 17.0 equivalent. 

<p float="left">
  
  
</p>

<p float="left">
  
  
</p>

### Steps to Reproduce

By adding the following code and then sliding the slider value, you'll see the editor will split the text into two lines on iOS 15.5 but not iOS 17+

```xaml
    <ScrollView>
        <VerticalStackLayout>
            <Slider
                x:Name="mySlider"
                Minimum="0.5"
                Maximum="1"
                Value="1"/>

            <Editor
                x:Name="myEditor"
                Text="1234567890"
                Scale="{Binding Value, Source={x:Reference mySlider}}"
                Background="Red"
                FontSize="32"
                HorizontalOptions="Center"
                HorizontalTextAlignment="Center" />

        </VerticalStackLayout>
    </ScrollView>
```

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.92 SR9.2

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15.5 - iOS 16.4

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Tamilarasan-Paranthaman**
### Root Cause of the issue 

- The layout of the Editor behaves differently on iOS 17.5 and earlier versions when the scale changes at runtime. Specifically, the content height gradually increases when scaling down the Editor, suggesting improper handling of height. 

### Description of Change

-  It seems that iOS 18.0 handles this behavior correctly. To ensure consistent behavior similar to iOS 18.0, I wrapped the Editor in a container for iOS 17.5 and earlier versions. This container ensures that the layout remains stable and visually consistent, even when the Editor is scaled or resized. Since the issue is resolved in iOS 18.0 and later, the additional container is not needed for those versions.

### Issues Fixed

Fixes #25581

### Tested the behaviour in the following platforms

- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac

### Screenshot

| Before Issue Fix | After Issue Fix |
|----------|----------|
| <video width="270" height="600"  src="https://github.com/user-attachments/assets/bde4b741-5cbe-49d4-89c6-546770c5a985"> | <video width="300" height="600"  src="https://github.com/user-attachments/assets/001f57da-855b-4106-9bf0-e9dc34b09395"> |

### CODE REVIEW

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Core/src/Handlers/Editor/EditorHandler.iOS.cs [hunk 1]
```diff
@@ -53,6 +53,23 @@ protected override void DisconnectHandler(MauiTextView platformView)
 			_proxy.Disconnect(platformView);
 		}
 
+		public override bool NeedsContainer
+		{
+			get
```
[the remark this answers is not in the document]
Comment (Tamilarasan-Paranthaman) [reply src/Core/src/Handlers/Editor/EditorHandler.iOS.cs:58]: @jsuarezruiz, I have added a UITest sample for this issue.

**Review by PureWeen — CHANGES_REQUESTED** [review]
Test failing on winui

**Review by jsuarezruiz — DISMISSED** [review]
The device tests:

```
ScrollEditor
Assert.True() Failure\nExpected: True\nActual:   False
```

and

```
ScrollNextEditor
Assert.True() Failure\nExpected: True\nActual:   False
```

are failing after the changes.

**Comment by Tamilarasan-Paranthaman** [comment]
> The device tests:
> 
> ```
> ScrollEditor
> Assert.True() Failure\nExpected: True\nActual:   False
> ```
> 
> and
> 
> ```
> ScrollNextEditor
> Assert.True() Failure\nExpected: True\nActual:   False
> ```
> 
> are failing after the changes.

@jsuarezruiz, I have fixed this by restricting the Mac platform in the fix, as the issue occurred on the iOS platform. The cases have passed locally. Could you please check and let me know if you have any concerns?

Screenshot

**Comment by tj-devel709** [comment]
@Tamilarasan-Paranthaman did you see this issue in iOS 17? I have only noticed it in 15.5 and 16.4 but maybe there is a scenario it happens in 17 also?

**Comment by Tamilarasan-Paranthaman** [comment]
> @Tamilarasan-Paranthaman did you see this issue in iOS 17? I have only noticed it in 15.5 and 16.4 but maybe there is a scenario it happens in 17 also?

@tj-devel709, I encountered this issue in iOS 17+ as well. Interestingly, I am currently unable to reproduce it in iOS 17+ (specifically tested on version 17.5). This is quite unusual, as the issue report also confirms it was [verified](https://github.com/dotnet/maui/issues/25581#issuecomment-2449448297) in version 17.5. Could you please share your thoughts on this and suggest how we should proceed?

**Comment by PureWeen** [comment]
@Tamilarasan-Paranthaman let's just go with what you're seeing

Can you just set the operating system version check to whatever version you find that we need this for?

**Comment by Tamilarasan-Paranthaman** [comment]
> @Tamilarasan-Paranthaman let's just go with what you're seeing
> 
> Can you just set the operating system version check to whatever version you find that we need this for?

@PureWeen, I have updated the version, and it will now apply only to versions below 17. Could you please check it now and let me know if you have any further concerns?

**Review by tj-devel709 — APPROVED** [review]
Confirmed with these new changes that it works with the following I versions!

18.1
18.0
17.5
17.2
17.01
16.4
15.5

**Review by PureWeen — APPROVED** [review]

**Pull request merged**